crates/containerd-shim-wasm/src/sandbox/cli.rs

```diff
@@ -37,7 +39,43 @@ macro_rules! revision {
     };
 }
 
-#[instrument(skip_all, parent = Span::current(), level= "Info")]
+/// Main entry point for the shim with OpenTelemetry tracing.
+///
+/// It parses the `OTEL_EXPORTER_OTLP_ENDPOINT` environment variable to determine
+/// if the shim should be started with OpenTelemetry tracing.
+///
+/// If the environment variable is not set, the shim will be started without tracing.
+/// If the environment variable is empty, the shim will be started without tracing.
+#[cfg(feature = "opentelemetry")]
+pub fn shim_main_with_otel<'a, I>(
+    name: &str,
+    version: &str,
+    revision: impl Into<Option<&'a str>>,
+    shim_version: impl Into<Option<&'a str>>,
+    config: Option<Config>,
+) where
+    I: 'static + Instance + Sync + Send,
+    I::Engine: Default,
+{
+    match std::env::var(OTEL_EXPORTER_OTLP_ENDPOINT) {
+        Ok(otel_endpoint) => {
+            let otel_config = OtelConfig::builder()
+                .otel_endpoint(otel_endpoint)
+                .name(name.to_string())
+                .build()
+                .expect("Failed to build OtelConfig.");
+            let _guard = otel_config
+                .init()
+                .expect("Failed to initialize OpenTelemetry.");
+            shim_main::<I>(name, version, revision, shim_version, config);
+        }
+        Err(_) => {
```

Comment:
should we log as a trace that the value wasn't set to help with debugging?

Reply:
Can't log a msg here since this will be called by the parent process while the logger is setup at the child process at a later time.